### PR TITLE
Add previous server to ServerConnectedEvent

### PR DESCRIPTION
**Pull request by Gabik21**

### CODE REVIEW

File: api/src/main/java/com/velocitypowered/api/event/player/ServerConnectedEvent.java [hunk 1]
```diff
@@ -26,6 +31,10 @@ public RegisteredServer getServer() {
     return server;
   }
 
+  public Optional<RegisteredServer> getPreviousServer() {
+    return Optional.ofNullable(previousServer);
+  }
+
   @Override
   public String toString() {
```
Comment (md678685) [inline api/src/main/java/com/velocitypowered/api/event/player/ServerConnectedEvent.java]: I believe `toString` should include `previousServer`.

File: api/src/main/java/com/velocitypowered/api/event/player/ServerConnectedEvent.java [hunk 2]
```diff
@@ -12,10 +14,19 @@
 
   private final Player player;
   private final RegisteredServer server;
+  private final @Nullable RegisteredServer previousServer;
 
-  public ServerConnectedEvent(Player player, RegisteredServer server) {
```
Comment (astei) [inline api/src/main/java/com/velocitypowered/api/event/player/ServerConnectedEvent.java]: API breakage. Readd this constructor and simply forward to the new one.

Comment (Gabik21) [reply api/src/main/java/com/velocitypowered/api/event/player/ServerConnectedEvent.java]: Should I mark it as deprecated?

Comment (astei) [reply api/src/main/java/com/velocitypowered/api/event/player/ServerConnectedEvent.java]: That's fine.

**Pull request merged**